app/src/main/java/com/github/siela1915/bootcamp/UploadingRecipeFragment.java

```diff
@@ -297,50 +306,55 @@ public void validate(TextView textView, String text) {
 
     public void setupIngredientAutocomplete(AutoCompleteTextView ingredientAutoComplete, Map<String, Integer> idMap, IngredientAutocomplete apiService){
         ingredientAutoComplete.setThreshold(1);
+
+        final BooleanWrapper optionSelected = new BooleanWrapper(false);
+        ingredientAutoComplete.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                // Clear the focus of the AutoCompleteTextView
+                ingredientAutoComplete.clearFocus();
+                optionSelected.setBool(true);
+            }
+        });
         ingredientAutoComplete.addTextChangedListener(new TextWatcher() {
             String prevString;
             boolean isTyping = false;
+            private final Handler handler = new Handler();
+            private final long DELAY = 500; // milliseconds
+
+
+            //function to be called if the user stops typing
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    if(!optionSelected.getBool()) {
```

Comment:
Here if you use a Boolean for optionSelected, no need for a function call just do !optionSelected